src/main/java/pl/betoncraft/betonquest/utils/location/CompoundLocation.java

```diff
@@ -0,0 +1,57 @@
+package pl.betoncraft.betonquest.utils.location;
+
+import org.bukkit.Location;
+import org.bukkit.util.Vector;
+import pl.betoncraft.betonquest.api.Variable;
+import pl.betoncraft.betonquest.config.ConfigPackage;
+import pl.betoncraft.betonquest.exceptions.InstructionParseException;
+import pl.betoncraft.betonquest.exceptions.QuestRuntimeException;
+
+/**
+ * This class parses various location related strings with or without {@link Variable}s.
+ */
+public class CompoundLocation {
+    private final LocationData locationData;
+    private final VectorData vectorData;
+
+    /**
+     * This class parses a string into a {@link Location} and a {@link Vector}. The input string has
+     * to be in the format 'x;y;z;world[;yaw;pitch][-> (x;y;z)]'. All elements in square brackets are optional.
+     * The last optional part is the {@link Vector} that will be added to the {@link Location} if specified.
+     * Each part of the input string can be a {@link Variable} instead of an {@link Integer} or {@link String}.
+     *
+     * @param packName Name of the {@link ConfigPackage} - required for {@link Variable} resolution
+     * @param data     string containing raw location in the defined format
+     * @throws InstructionParseException Is thrown when an error appears while parsing {@link LocationData} or {@link VectorData}
+     */
+    public CompoundLocation(final String packName, final String data) throws InstructionParseException {
```

Comment:
Create a second constructor to initialize a `CompoundLocation` with a `LocationData` and an optional `VectorData` instance instead of always forcing re-parsing?

Reply:
we do not have a usecase for that